acceptance/suites/pre_suite/foss/30_install_dev_repos.rb

```diff
@@ -1,27 +1,19 @@
-install_opts = options.merge( { :dev_builds_repos => ["PC1"] })
-repo_config_dir = 'tmp/repo_configs'
+if test_config[:puppetserver_install_type] == :package
+  package_build_version = ENV['PACKAGE_BUILD_VERSION']
+  if package_build_version.nil?
+    abort("Environment variable PACKAGE_BUILD_VERSION required for package installs!")
+  end
 
-case test_config[:puppetserver_install_type]
-when :package
-  step "Setup Puppet Server repositories." do
-    package_build_version = ENV['PACKAGE_BUILD_VERSION']
-    if package_build_version
-      install_puppetlabs_dev_repo master, 'puppetserver', package_build_version,
-                                  repo_config_dir, install_opts
-    else
-      abort("Environment variable PACKAGE_BUILD_VERSION required for package installs!")
-    end
+  step "Setup Puppet Server dev repository on the master." do
+    install_opts = options.merge({ :dev_builds_repos => ["PC1"]})
+    install_puppetlabs_dev_repo(master, 'puppetserver', package_build_version, nil, install_opts)
   end
 end
 
-puppet_build_version = test_config[:puppet_build_version]
-if puppet_build_version
-  confine_block :except, :platform => ['windows'] do
-    step "Setup Puppet Labs Dev Repositories." do
-      hosts.each do |host|
-        install_puppetlabs_dev_repo host, 'puppet-agent', puppet_build_version,
-                                    repo_config_dir, install_opts
-      end
+confine_block :except, :platform => ['windows'] do
+  step "Setup Puppet Labs Dev Repositories." do
+    hosts.each do |host|
+      install_puppet_agent_dev_repo_on(host, {:puppet_agent_version => test_config[:puppet_build_version]})
```

Comment:
Looks like the puppetserver nightlies failed CI on SLES 12 only after this change went in.  Appears to be trying to access the wrong URL to download the rpm which gets installed:

```
Setup Puppet Labs Dev Repositories.
Fetching: http://builds.delivery.puppetlabs.net/puppet-agent/e749206bc4be4c81b604c305ce7550bf36f54524/repos/sles/12/PC1/x86_64/puppet-agent-e749206bc4be4c81b604c305ce7550bf36f54524-1.sles12.x86_64.rpm
```

I think the problem is due to the `install_puppet_agent_dev_repo_on` function falling back to trying to download the rpm directly instead of using the repo_config, as it was prior to that change, and failing.  See [this code](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/install_utils/foss_utils.rb#L1114-L1135).  We might be able to make the .rpm file resolve correctly if we change this to pass in both a `puppet_agent_sha` and `puppet_agent_version`.  Having to maintain both in our helper config would seem onerous, though.  We might also be able to change beaker to use a repo_config instead of an rpm like is done for other EL variants but I'm not sure if that would have broader implications.  Maybe it would be best to just revert this back to using `install_puppetlabs_dev_repo` as it was previously?